provider/blob/src/blob_data_provider.rs

```diff
@@ -69,6 +69,12 @@ impl BlobDataProvider {
             })?,
         })
     }
+
+    #[cfg(feature = "export")]
+    #[doc(hidden)] // See #1771, we don't want this to be a publicly visible API
+    pub fn get_map(&self) -> &ZeroMap2dBorrowed<ResourceKeyHash, str, [u8]> {
```

Comment:
Thought: You know how I feel about making BlobDataProvider be iterable. At least this is hidden and used only by test code, but I have a slight preference for holding this by about a week, at which point ResourceOptions will be much-improved and we can use standard APIs for this.

Reply:
Sounds good, though personally I'd prefer to land this and then use this code as the first client to test out iterable resource options

Reply:
ok.